### DISCUSSION TITLE
DCP rule

### DISCUSSION THREAD

**Discussion by yanalish** [cvxpy/cvxpy#1813] [Q&A] [answered]
Hi,

Would anyone be able to  explain to me why the expression -cvxpy.exp(-x)@x is non DCP? Is there a way in which I could define this expression to pass the DCP rule?

Thank you very much in advance!

**Comment by SteveDiamond** [comment, marked as answer]
There is an xexp atom that is relevant here: https://www.cvxpy.org/_modules/cvxpy/atoms/elementwise/xexp.html

**Comment by yanalish** [comment]
Hi, 

Thank you very much! I was on the old version thus I did not know this had been added. I haven't tried it as I am having issues updating all the required packages due to other dependencies, but I am sure this was the solution to my question! Thank you very much for your help and the quick response!
